performances/performance_test/include/performance_test/ros2/node.hpp

```diff
@@ -347,15 +348,21 @@ friend class System;
     _client_lock = true;
 
     // Get client and tracking count from map
-    auto& client_pair = _clients.at(name);
-    auto client = std::static_pointer_cast<rclcpp::Client<Srv>>(client_pair.first);
-    auto& tracker = client_pair.second;
+    auto& client_tuple = _clients.at(name);
+    auto client = std::static_pointer_cast<rclcpp::Client<Srv>>(std::get<0>(client_tuple));
+    auto& tracker = std::get<1>(client_tuple);
+    auto& tracking_number = std::get<2>(client_tuple);
+
+    //Wait for service to come online
+    while (!client->wait_for_service(1.0s)){
+      RCLCPP_WARN(this->get_logger(), "service '%s' not available, waiting again...", name.c_str());
```

Comment:
This will block the whole node.
Maybe a better solution is to simply return from this function if the server is not alive.
The check will be performed again the next time a request has to be sent. 

The fact that the server is not alive may be logged as an event using the `_event_logger` and creating a new event type for this.